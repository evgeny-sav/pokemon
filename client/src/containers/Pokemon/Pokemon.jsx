import React, { useEffect } from 'react';
import { useRouteMatch } from 'react-router-dom';
import { useDispatch, useSelector, shallowEqual } from 'react-redux';
import { Container } from 'react-bootstrap';

import Loading from '../../components/Loading';
import StatsTable from '../../components/StatsTable';
import Slider from '../../components/Slider';
import GoBackButton from '../../components/GoBackButton';
import capitalize from '../../utils/capitalize';
import { fetchPokemon } from '../../store/actions/singlePokemonActions';
import './Pokemon.scss';

const Pokemon = () => {
  const dispatch = useDispatch();
  const { url } = useRouteMatch();
  const { isLoading, pokemon } = useSelector(state => {
    return {
      isLoading: state.loading.isLoading,
      pokemon: state.pokemon.data,
    };
  }, shallowEqual);
  const { name } = pokemon;

  useEffect(() => {
    dispatch(fetchPokemon(url));
  }, []);

  return isLoading ? (
    <Loading />
  ) : (
    <Container>
      <div className="Pokemon">
        <div className="goback">
          <GoBackButton />
        </div>

        <div className="content">
          <div className="slider">
            <Slider />
          </div>

          <div className="info">
            <h3 className="mb-5">{capitalize(name)}</h3>
            <StatsTable pokemon={pokemon} />
          </div>
        </div>
      </div>
    </Container>
  );
};

export default Pokemon;
