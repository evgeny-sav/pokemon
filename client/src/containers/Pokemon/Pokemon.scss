@import "src/colors";

.Pokemon {
  display: flex;
  height: calc(100vh - 200px);
  flex-direction: column;

  .content {
    display: flex;
    flex-direction: row;

    .slider {
      display: flex;
      width: 100%;
      height: 200px;
      justify-content: center;
      margin-bottom: 100px;
    }

    .info {
      display: flex;
      flex-direction: column;
      width: 100%;
    }

    @media (max-width: 768px) {
      flex-direction: column;
    }

  }

  .carousel-inner {
    width: 375px;
  }

  .carousel-indicators {
    bottom: -50px;
    li {
      background-color: $yellow;
    }
  }
  .carousel-control-prev, .carousel-control-next {
    color: $yellow;
  }

  .carousel-control-next-icon {
    background: $yellow;
  }
}
