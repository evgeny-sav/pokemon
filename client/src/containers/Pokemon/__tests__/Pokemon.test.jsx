import React from 'react';
import { create } from 'react-test-renderer';
import Loading from '../../../components/Loading';
import Pokemon from '../Pokemon';

jest.mock('../../../components/Loading', () => 'Loading');
jest.mock('../../../components/StatsTable', () => 'StatsTable');
jest.mock('../../../components/Slider', () => 'Slider');
jest.mock('../../../components/GoBackButton', () => 'GoBackButton');

jest.mock('react-router-dom', () => ({
  useRouteMatch: () => jest.fn(),
}));

jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
  useSelector: jest
    .fn()
    .mockImplementationOnce(() => {
      return {
        isLoading: true,
        pokemon: {},
      };
    })
    .mockImplementationOnce(() => {
      return {
        isLoading: false,
        pokemon: {
          name: 'name',
        },
      };
    }),
}));

describe('Pokemon', () => {
  it('should render loading', () => {
    const component = create(<Pokemon />);
    expect(component.root.findByType(Loading).type).toBe('Loading');
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render with pokemon data', () => {
    const component = create(<Pokemon />);
    expect(component.toJSON().children[0].props.className).toEqual('Pokemon');
    expect(component.toJSON()).toMatchSnapshot();
  });
});
