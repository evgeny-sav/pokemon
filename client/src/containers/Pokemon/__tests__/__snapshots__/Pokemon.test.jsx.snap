// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon should render loading 1`] = `<Loading />`;

exports[`Pokemon should render with pokemon data 1`] = `
<div
  className="container"
>
  <div
    className="Pokemon"
  >
    <div
      className="goback"
    >
      <GoBackButton />
    </div>
    <div
      className="content"
    >
      <div
        className="slider"
      >
        <Slider />
      </div>
      <div
        className="info"
      >
        <h3
          className="mb-5"
        >
          Name
        </h3>
        <StatsTable
          pokemon={
            Object {
              "name": "name",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
